src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,406 @@
+	MangaViewer, Page, std::current_date, helpers::uri::encode_uri_component
+};
+
+pub fn parse_manga_list(base_url: String, filters: Vec<Filter>, _page: i32) -> Result<MangaPageResult> {
+	
+	let mut search_query = String::new();
+	let mut status = String::new();
+	let mut genres = String::new();
+	
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					search_query.push_str(&encode_uri_component(filter_value.read()));
+				}
+			}
+			FilterType::Genre => {
+				if let Ok(filter_id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						1 => {
+							genres.push_str(filter_id.read().as_str());
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => status.push_str("All"),
+					1 => status.push_str("Ongoing"),
+					2 => status.push_str("Hiatus"),
+					3 => status.push_str("Dropped"),
+					4 => status.push_str("Completed"),
+					5 => status.push_str("Cancelled"),
+					_ => continue,
+				},
+				_ => continue,
+			}
+			_ => continue,
+		}
+	}
+
+	if !genres.is_empty() {
+		genres.pop();
+	}
+
+	let url = format!("https://api.omegascans.org/query?query_string={}&series_status={}&order=desc&orderBy=total_views&series_type=Comic&page=1&perPage=1000&tags_ids=[{}]", search_query, status, genres);
+
+	let json = Request::new(url, HttpMethod::Get)
+		.json()
+		.expect("Failed to load JSON")
+		.as_object()
+		.expect("Failed to get JSON as object");
+
+	let data = json
+		.get("data")
+		.as_array()
+		.expect("Failed to get data as object");
+
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	for manga in data {
+		let manga = manga.as_object().expect("Failed to get manga as object");
+
+		let title = manga
+			.get("title")
+			.as_string()
+			.expect("Failed to get manga name as str")
+			.read();
+
+		let cover = manga
+			.get("thumbnail")
+			.as_string()
+			.expect("Failed to get manga cover as str")
+			.read();
+
+		let slug = manga
+			.get("series_slug")
+			.as_string()
+			.expect("Failed to get manga slug as str")
+			.read();
+
+		let status = manga
+			.get("status")
+			.as_string()
+			.expect("Failed to get manga status as array")
+			.read();
+
+		let mut manga_status = MangaStatus::Unknown;
+
+		match status.as_str() {
+			"New" => {}
+			"Ongoing" => manga_status = MangaStatus::Ongoing,
+			"Completed" => manga_status = MangaStatus::Completed,
+			"Dropped" => manga_status = MangaStatus::Cancelled,
+			"Hiatus" => manga_status = MangaStatus::Hiatus,
+			_ => manga_status = MangaStatus::Unknown,
+		};
+
+
+		let url = format!("{}/series/{}", base_url, slug);
+
+		mangas.push(Manga {
+			id: slug,
+			cover: cover,
+			title: title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
```

Comment:
use `..Default::default()` at the end of the struct to populate default values instead of including your own